### ISSUE TITLE
cannot import mp3 any more

### ISSUE THREAD

**Issue by bassman56** [audacity/audacity#5617]
### Bug description

I cannot import mp3ś any more (on 3.4.1, Linux, opensuse tumbleweed). it worked on 3.3.3 with the same mp3's.
In the settings, the "library" tag is missing where usually the links to ffmpeg are  stored.

### Steps to reproduce

start audacity
open any mp3
get an error message that the file type is not known(in german).
Open "settings": there is no tab for libraries ("Bibliotheken") any more, which was present in 3.3.3

### Expected behavior

an mp3 file should be opened.
the tab for the ffmpeg libraries shoudl be present

### Actual behavior

see above, mp3 cannot be opened

### Audacity Version

stable version (from a Linux repository)

### Operating system

Linux

**Comment by LWinterberg** [comment]
sounds like there's something broken with your installation. Try using the appimage from audacityteam.org instead.

**Comment by LWinterberg** [comment]
There should be a dozen or so files handling imports in /usr/lib64/audacity/modules/. Do you see them?

**Comment by bassman56** [comment]
I see only 1 file  /usr/lib64/audacity/modules/mod-scipt-pipe.po
I now will roll back to old system status 3 days ago, freeze audacity, update all other modules and try a selective update of audacity once more.

**Comment by bassman56** [comment]
how to use an app-image in opensuse tumbleweed?

**Comment by LWinterberg** [comment]
in principle you just mark the appimage as an executable and then it should be able to run standalone.

**Comment by LWinterberg** [comment]
CC @plater probably - there may be some files missing in your audacity package

**Comment by bassman56** [comment]
What about the missing settings for library where the ffmpegs are to be located? is this intended with 3.4.1 (vs 3.3.3)?

**Comment by LWinterberg** [comment]
yep, that looks just about right. We don't expose these modules in the UI at the moment as they're more on the "technically optional" side than "actually useful to manually disable" side.

**Comment by bassman56** [comment]
but I still cannot import mp3's....

**Comment by vsverchinsky** [comment]
> We don't expose these modules in the UI at the moment as they're more on the "technically optional" side than "actually useful to manually disable" side.

Actually, we do list all modules, but enable all exporting modules by default (just noticed that text says that they are experimental)
![изображение](https://github.com/audacity/audacity/assets/20165230/b6973f2a-3657-4880-b95c-ee71dd4a519f)
@bassman56 could you please check that all exporting plugins are available on that page and enabled?

**Comment by plater** [comment]
I don't have any problem with importing mp3s in Audacity. Could the reporter execute from the command line:
"ldd /usr/bin/audacity |grep lame", "ldd /usr/bin/audacity |grep mpg" and "ldd /usr/bin/audacity |grep mpeg"
This should expose  /usr/lib64/libtwolame.so.0, /usr/lib64/libmpg123.so.0 and /usr/lib64/audacity/lib-ffmpeg-support.so

If these libraries are somehow missing that would explain the problem.
There's only one file in /usr/lib64/audacity/modules and that is mod-script-pipe.so

**Comment by bassman56** [comment]
Hi plater,
all 3 ldd commands show nothing, so something is missing? What do I need to install? lame itself is present as well as ffmpeg. 
the library is present as well:
ls -l /usr/lib64/libtwolame.so.0
lrwxrwxrwx 1 root root 19 Aug 12 17:43 /usr/lib64/libtwolame.so.0 -> libtwolame.so.0.0.0
ls -l  /usr/lib64/libmpg123.so.0
lrwxrwxrwx 1 root root 19 Oct  4 22:48 /usr/lib64/libmpg123.so.0 -> libmpg123.so.0.48.1
ls -l  /usr/lib64/audacity/lib-ffmpeg-support.so
-rwxr-xr-x 1 root root 325448 Nov 10 13:13 /usr/lib64/audacity/lib-ffmpeg-support.so

**Comment by crsib** [comment]
> There's only one file in /usr/lib64/audacity/modules and that is mod-script-pipe.so
> "ldd /usr/bin/audacity |grep mpg"

This has changed in 3.4; importers and exporters are now distributed as modules. Audacity no longer links directly to any of the codecs.

**Comment by bassman56** [comment]
I found a workaround by stracing audacity: it seems that audacity looks for the modules in /usr/lib/audacity . But actually the files are in /usr/lib64/audacity.  So I created a symbolic link in /usr/lib as follows:
/usr/lib>  ln -s /usr/lib64/audacity/ .

Now the import of mp3s works again, most likely other stuff works as well(need to try).  
Who needs to fix this: the opensuse tumbleweed distro or the audacity sources?

**Comment by crsib** [comment]
Probably it should be fixed from our side. I took a look and Arch installs Audacity libraries into `/usr/lib/audacity`, and it has a very different structure overall. 

```
dvedenko@crsib-desktop ~> ls -la /usr/ | grep lib
drwxr-xr-x 121 root root 98304 Nov 14 12:56 lib
drwxr-xr-x   2 root root  4096 Aug  7 16:38 lib32
lrwxrwxrwx   1 root root     3 Sep 18 16:18 lib64 -> lib
```

I will check Ubuntu as well

**Comment by degrees57** [comment]
> I found a workaround by stracing audacity: it seems that audacity looks for the modules in /usr/lib/audacity . But actually the files are in /usr/lib64/audacity. So I created a symbolic link in /usr/lib as follows: /usr/lib> ln -s /usr/lib64/audacity/ .
> 
> Now the import of mp3s works again, most likely other stuff works as well(need to try). Who needs to fix this: the opensuse tumbleweed distro or the audacity sources?

THANK YOU!  This solved the same problem for me, too.

**Comment by plater** [comment]
The users with audacity's libraries on openSUSE which are under /usr/lib/audacity either have a foreign package installed or they have inadvertently installed an i586 32 bit package. Here's the contents of the official package:
rpm -ql audacity
/usr/bin/audacity
/usr/lib64/audacity
/usr/lib64/audacity/lib-audio-devices.so
/usr/lib64/audacity/lib-audio-graph.so
/usr/lib64/audacity/lib-audio-io.so
/usr/lib64/audacity/lib-basic-ui.so
/usr/lib64/audacity/lib-channel.so
/usr/lib64/audacity/lib-command-parameters.so
/usr/lib64/audacity/lib-components.so
/usr/lib64/audacity/lib-effects.so
/usr/lib64/audacity/lib-exceptions.so
/usr/lib64/audacity/lib-export-ui.so
/usr/lib64/audacity/lib-ffmpeg-support.so
/usr/lib64/audacity/lib-file-formats.so
/usr/lib64/audacity/lib-files.so
/usr/lib64/audacity/lib-graphics.so
/usr/lib64/audacity/lib-import-export.so
/usr/lib64/audacity/lib-ipc.so
/usr/lib64/audacity/lib-ladspa.so
/usr/lib64/audacity/lib-lv2.so
/usr/lib64/audacity/lib-math.so
/usr/lib64/audacity/lib-mixer.so
/usr/lib64/audacity/lib-module-manager.so
/usr/lib64/audacity/lib-numeric-formats.so
/usr/lib64/audacity/lib-playable-track.so
/usr/lib64/audacity/lib-preferences.so
/usr/lib64/audacity/lib-project-file-io.so
/usr/lib64/audacity/lib-project-history.so
/usr/lib64/audacity/lib-project-rate.so
/usr/lib64/audacity/lib-project.so
/usr/lib64/audacity/lib-realtime-effects.so
/usr/lib64/audacity/lib-registries.so
/usr/lib64/audacity/lib-sample-track.so
/usr/lib64/audacity/lib-screen-geometry.so
/usr/lib64/audacity/lib-shuttlegui.so
/usr/lib64/audacity/lib-snapping.so
/usr/lib64/audacity/lib-stretching-sequence.so
/usr/lib64/audacity/lib-string-utils.so
/usr/lib64/audacity/lib-strings.so
/usr/lib64/audacity/lib-tags.so
/usr/lib64/audacity/lib-theme-resources.so
/usr/lib64/audacity/lib-theme.so
/usr/lib64/audacity/lib-time-and-pitch.so
/usr/lib64/audacity/lib-time-frequency-selection.so
/usr/lib64/audacity/lib-time-track.so
/usr/lib64/audacity/lib-track-selection.so
/usr/lib64/audacity/lib-track.so
/usr/lib64/audacity/lib-transactions.so
/usr/lib64/audacity/lib-utility.so
/usr/lib64/audacity/lib-uuid.so
/usr/lib64/audacity/lib-vst.so
/usr/lib64/audacity/lib-vst3.so
/usr/lib64/audacity/lib-wave-track.so
/usr/lib64/audacity/lib-wx-init.so
/usr/lib64/audacity/lib-wx-wrappers.so
/usr/lib64/audacity/lib-xml.so
/usr/lib64/audacity/modules
/usr/lib64/audacity/modules/mod-aup.so
/usr/lib64/audacity/modules/mod-cl.so
/usr/lib64/audacity/modules/mod-ffmpeg.so
/usr/lib64/audacity/modules/mod-flac.so
/usr/lib64/audacity/modules/mod-lof.so
/usr/lib64/audacity/modules/mod-mp2.so
/usr/lib64/audacity/modules/mod-mp3.so
/usr/lib64/audacity/modules/mod-mpg123.so
/usr/lib64/audacity/modules/mod-ogg.so
/usr/lib64/audacity/modules/mod-opus.so
/usr/lib64/audacity/modules/mod-pcm.so
/usr/lib64/audacity/modules/mod-script-pipe.so
/usr/lib64/audacity/modules/mod-wavpack.so
/usr/share/applications/audacity.desktop
/usr/share/audacity
/usr/share/audacity/EffectsMenuDefaults.xml
/usr/share/audacity/nyquist
/usr/share/audacity/nyquist/aud-do-support.lsp
/usr/share/audacity/nyquist/dspprims.lsp
/usr/share/audacity/nyquist/envelopes.lsp
/usr/share/audacity/nyquist/equalizer.lsp
/usr/share/audacity/nyquist/evalenv.lsp
/usr/share/audacity/nyquist/fileio.lsp
/usr/share/audacity/nyquist/init.lsp
/usr/share/audacity/nyquist/misc.lsp
/usr/share/audacity/nyquist/nyinit-dbg.lsp
/usr/share/audacity/nyquist/nyinit.lsp
/usr/share/audacity/nyquist/nyqmisc.lsp
/usr/share/audacity/nyquist/nyquist-plot.txt
/usr/share/audacity/nyquist/nyquist.lsp
/usr/share/audacity/nyquist/printrec.lsp
/usr/share/audacity/nyquist/profile.lsp
/usr/share/audacity/nyquist/rawwaves
/usr/share/audacity/nyquist/rawwaves/mand1.raw
/usr/share/audacity/nyquist/rawwaves/mand10.raw
/usr/share/audacity/nyquist/rawwaves/mand11.raw
/usr/share/audacity/nyquist/rawwaves/mand12.raw
/usr/share/audacity/nyquist/rawwaves/mand2.raw
/usr/share/audacity/nyquist/rawwaves/mand3.raw
/usr/share/audacity/nyquist/rawwaves/mand4.raw
/usr/share/audacity/nyquist/rawwaves/mand5.raw
/usr/share/audacity/nyquist/rawwaves/mand6.raw
/usr/share/audacity/nyquist/rawwaves/mand7.raw
/usr/share/audacity/nyquist/rawwaves/mand8.raw
/usr/share/audacity/nyquist/rawwaves/mand9.raw
/usr/share/audacity/nyquist/rawwaves/mandpluk.raw
/usr/share/audacity/nyquist/rawwaves/marmstk1.raw
/usr/share/audacity/nyquist/rawwaves/sinewave.raw
/usr/share/audacity/nyquist/sal-parse.lsp
/usr/share/audacity/nyquist/sal.lsp
/usr/share/audacity/nyquist/seq.lsp
/usr/share/audacity/nyquist/seqfnint.lsp
/usr/share/audacity/nyquist/seqmidi.lsp
/usr/share/audacity/nyquist/sliders.lsp
/usr/share/audacity/nyquist/sndfnint.lsp
/usr/share/audacity/nyquist/spec-plot.lsp
/usr/share/audacity/nyquist/spectral-analysis.lsp
/usr/share/audacity/nyquist/stk.lsp
/usr/share/audacity/nyquist/system.lsp
/usr/share/audacity/nyquist/test.lsp
/usr/share/audacity/nyquist/velocity.lsp
/usr/share/audacity/nyquist/xlinit.lsp
/usr/share/audacity/nyquist/xm.lsp
/usr/share/audacity/plug-ins
/usr/share/audacity/plug-ins/ShelfFilter.ny
/usr/share/audacity/plug-ins/SpectralEditMulti.ny
/usr/share/audacity/plug-ins/SpectralEditParametricEQ.ny
/usr/share/audacity/plug-ins/SpectralEditShelves.ny
/usr/share/audacity/plug-ins/StudioFadeOut.ny
/usr/share/audacity/plug-ins/adjustable-fade.ny
/usr/share/audacity/plug-ins/beat.ny
/usr/share/audacity/plug-ins/clipfix.ny
/usr/share/audacity/plug-ins/crossfadeclips.ny
/usr/share/audacity/plug-ins/crossfadetracks.ny
/usr/share/audacity/plug-ins/delay.ny
/usr/share/audacity/plug-ins/eq-xml-to-txt-converter.ny
/usr/share/audacity/plug-ins/equalabel.ny
/usr/share/audacity/plug-ins/highpass.ny
/usr/share/audacity/plug-ins/label-sounds.ny
/usr/share/audacity/plug-ins/limiter.ny
/usr/share/audacity/plug-ins/lowpass.ny
/usr/share/audacity/plug-ins/noisegate.ny
/usr/share/audacity/plug-ins/notch.ny
/usr/share/audacity/plug-ins/nyquist-plug-in-installer.ny
/usr/share/audacity/plug-ins/pluck.ny
/usr/share/audacity/plug-ins/rhythmtrack.ny
/usr/share/audacity/plug-ins/rissetdrum.ny
/usr/share/audacity/plug-ins/rms.ny
/usr/share/audacity/plug-ins/sample-data-export.ny
/usr/share/audacity/plug-ins/sample-data-import.ny
/usr/share/audacity/plug-ins/spectral-delete.ny
/usr/share/audacity/plug-ins/tremolo.ny
/usr/share/audacity/plug-ins/vocalrediso.ny
/usr/share/audacity/plug-ins/vocoder.ny
/usr/share/doc/packages/audacity
/usr/share/doc/packages/audacity/README.md
/usr/share/icons/hicolor/16x16
/usr/share/icons/hicolor/16x16/audacity.png
/usr/share/icons/hicolor/22x22
/usr/share/icons/hicolor/22x22/audacity.png
/usr/share/icons/hicolor/24x24
/usr/share/icons/hicolor/24x24/audacity.png
/usr/share/icons/hicolor/32x32
/usr/share/icons/hicolor/32x32/audacity.png
/usr/share/icons/hicolor/48x48
/usr/share/icons/hicolor/48x48/audacity.png
/usr/share/icons/hicolor/48x48/mimetypes
/usr/share/icons/hicolor/48x48/mimetypes/application-x-audacity-project.xpm
/usr/share/icons/hicolor/scalable
/usr/share/icons/hicolor/scalable/apps
/usr/share/icons/hicolor/scalable/apps/audacity.svg
/usr/share/licenses/audacity
/usr/share/licenses/audacity/LICENSE.txt
/usr/share/licenses/audacity/LICENSE_NYQUIST.txt
/usr/share/man/man1/audacity.1.gz
/usr/share/metainfo
/usr/share/metainfo/audacity.appdata.xml
/usr/share/mime/packages/audacity.xml

**Comment by plater** [comment]
> I found a workaround by stracing audacity: it seems that audacity looks for the modules in /usr/lib/audacity . But actually the files are in /usr/lib64/audacity. So I created a symbolic link in /usr/lib as follows: /usr/lib> ln -s /usr/lib64/audacity/ .
> 
> Now the import of mp3s works again, most likely other stuff works as well(need to try). Who needs to fix this: the opensuse tumbleweed distro or the audacity sources?

I'm pretty sure you have a 32 bit audacity installed on a 64 bit system and you may have more waiting to create havoc with other apps. Try executingrpm -qa |grep i586
I did this on my system and now I know where the strange installation  problems with libx264 came from, I found:
libx264-161-0.161+git20200912.d198931a-3.19.i586
This I've fixed with:
zypper rm -u libx264-161-0.161+git20200912.d198931a-3.19.i586
I would suggest you execute "zypper rm -u audacity" and then after a refresh zypper -v in audacity and make sure that the x86_64 package is installed then audacity will work correctly

**Comment by plater** [comment]
A complaint about the audacity libraries under /usr/lib64/audacity. I have to break packaging rules and run ldconfig on /usr/lib64/audacity otherwise audacity doesn't find it's own libraries, normally programs which have libraries in a private directory use dlopen to use them.

**Comment by crsib** [comment]
> libraries in a private directory use dlopen to use them.

That's precisely the reason for this bug. Audacity uses `dlopen` and fails to find it in the expected place.

> run ldconfig on /usr/lib64/audacity 

And this one is strange. A similar problem was solved for Debian-based systems. I wonder why it doesn't work for openSUSE. Fedora doesn't run ldconfig either.

The modules do not work on Fedora though, I will take a look how the search path is constructed.

**Comment by bassman56** [comment]
Hi plater, thanks for the advice. I checked for i586 packages, there was only one unrelated  ibvapoursynth-53-53-40.1.i586
which I deleted. the installed audacity packet is audacity-3.4.1-2.2.x86_64 on tumbleweed from yesterday.  I will now try to uninstall and re-install, deleting the link /usr/lib/audacity (pointing to ../lib64). please standby

**Comment by bassman56** [comment]
I did the re-install 
Selecting 'audacity-3.4.1-2.2.x86_64' from repository 'Haupt-Repository (OSS) Tumbleweed' for installation.
Resolving package dependencies...
Force resolution: No

The following recommended package was automatically selected:
  audacity-lang  3.4.1-2.2

The following 8 NEW packages are going to be installed:
  audacity                          3.4.1-2.2
  audacity-lang                     3.4.1-2.2
  libwx_baseu-suse-nostl9_0_0       3.2.2.1-3.4
  libwx_baseu_net-suse-nostl9_0_0   3.2.2.1-3.4
  libwx_baseu_xml-suse-nostl9_0_0   3.2.2.1-3.4
  libwx_gtk3u_core-suse-nostl9_0_0  3.2.2.1-3.4
  libwx_gtk3u_html-suse-nostl9_0_0  3.2.2.1-3.4
  libwx_gtk3u_qa-suse-nostl9_0_0    3.2.2.1-3.4

8 new packages to install.
Overall download size: 12.1 MiB. Already cached: 0 B. After the operation, additional 47.6 MiB will be used.
Continue? [y/n/v/...? shows all options] (y): y
committing...
Result:
I still need the link /usr/lib/audacity --> /usr/lib64/audacity  to enable audacity to read mp3 file. If I delete the link, audacity reports not to recognize the file type.

Any advice how to continue?

**Comment by crsib** [comment]
I have addressed the module search paths in #5631, which is scheduled to be released with 3.4.2. 3.4.2 will be finished after the QA for it is complete. 

A quick workaround for now would be to set `AUDACITY_MODULES_PATH` env variable to `/usr/lib64/audacity`

**Comment by bassman56** [comment]
Hi crsib,
thanks for the advice, but the env needs to point to /usr/lib64/audacity/**modules** 
I wrote a short shell script to start audacity ---- works now :-) 
Thanks
